cogctl/cli/config.py

```diff
@@ -1,108 +1,109 @@
+import copy
+import os
+from configobj import ConfigObj
 from collections import OrderedDict
-from configparser import ConfigParser
 
 
-def read_config(file):
-    return ini_file_to_map(file)
+class CogctlConfig():
 
-
-def add_profile(file, profile_name, profile):
-    # Use OrderedDict just to ensure that the individual entries (in
-    # an INI file, at any rate) are always written out
-    # deterministically. If nothing else, this makes testing a bit
-    # easier.
-    new_profile = OrderedDict([
-        ("host", profile.pop("host")),
-        ("port", profile.pop("port")),
-        ("secure", profile.pop("secure")),
-        ("user", profile.pop("user")),
-        ("password", profile.pop("password"))])
-
-    return add_ini_section(file, profile_name, new_profile)
-
-########################################################################
-
-
-def read_file(ini):
-    config = ConfigParser(comment_prefixes=("#"),
-                          interpolation=None)
-    config.read(ini)
-    return config  # TODO: return type here (ini or netrc)
-
-
-def ini_file_to_map(ini):
-    """Convert an INI-formatted file into a plain dict"""
-
-    config = read_file(ini)
-
-    data = {}
-    default = None
-
-    for section in config.sections():
-        if section == "defaults":
-            default = config[section]["profile"]
+    def __init__(self, filename):
+        self.filename = filename
+        if os.path.isfile(filename):
+            # If the file exists it should be valid, so just try to
+            # get the default profile name and the default profile
+            self._config = ConfigObj(filename)
+            self.default_profile_name()
+            self.default()
```

Comment:
If I'm understanding this bit properly, your just doing this to see if an error is thrown. Do you need the call to `self.default_profile_name()` since `self.default()` makes that call anyway?